mathics/core/evaluation.py

```diff
@@ -117,10 +149,11 @@ def run_with_timeout_and_stack(request, timeout, evaluation):
     thread.join(timeout)
     if thread.is_alive():
         evaluation.timeout = True
-        while thread.is_alive():
-            pass
-        evaluation.timeout = False
+        time.sleep(0.01)
+        if not kill_thread(thread):
+            print("thread couldn't be stopped.")
```

Comment:
I wonder if there is something more appropriate than `print`?

Reply:
Sure, but this was supposed to be a draft. Tonmorow I am going to replace this with a message.

Reply:
> > Now it seems that also works in pyston.
> 
> That reminds me, it would be nice to have another gitflow that _demonstrates_ that this works in pyston. And maybe one for Python 3. 10 as well.

@mmatera are you up for adding this?

Reply:
Yes, but I am not sure right now how to do it.

Reply:
Ok - I will do. I have an opinion on this too. 

The main thing is to do this as a separate workflow so we don't increase the overall time to completion. And since ubuntu is the fastest and doesn't seem to have quotas on the number of runs (I guess partly because it is a free OS so the CI provider can spin up an unlimited number of these), this suggests we do this like we do in Cython in Ubuntu only.  

In my opinion we get value in testing say Python 3.6, whether there is value in testing Python 3.6 across *all* platforms is questionable. 

And down the line maybe we should switch to pyston across all platforms, especially Windows where testing takes the longest.